app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt

```diff
@@ -74,7 +75,14 @@ class PatientSearchResultsScreen(context: Context, attrs: AttributeSet) : Relati
     return searchResultsView
         .upstreamUiEvents
         .ofType<RegisterNewPatient>()
-        .map { PatientSearchResultRegisterNewPatient(it.patientName) }
+        .map { PatientSearchResultRegisterNewPatient(extractPatientName(it.searchBy)) }
+  }
+
+  private fun extractPatientName(searchBy: SearchPatientBy): String {
+    return when (searchBy) {
+      is SearchPatientBy.Name -> searchBy.searchText
+      is SearchPatientBy.PhoneNumber -> TODO("not yet implemented")
```

Comment:
Is there a possibility for this branch to get executed at the moment?

Reply:
Nope, the screen this view is embedded in will only invoke this with the name criteria at the moment.